src/lib.rs

```diff
@@ -51,6 +58,9 @@ pub trait AsHandle {
 }
 
 #[macro_export]
+/// Macro for calling `c-style functions` and wrapping the return status in a `Result`
+/// If the function return `-1` it will return `Err<$err:expr>` otherwise `Ok(int)`
+// FIXME: this is not platform independent: `https://github.com/paritytech/parity-daemonize/issues/14`
 macro_rules! map_err {
```

Comment:
Why do we expose this macro in the first place, can we move to unix submodule/or add `#[doc(hidden)]`?

Reply:
I don't know and I think we shouldn't export it but I would prefer keep it out of this PR.
I can mark it as `#[doc(hidden)] ok?

Also I think we should write this better and not predict error cause, such as Rust does it: https://github.com/rust-lang/rust/blob/master/src/libstd/sys/unix/mod.rs#L129-#L135

Reply:
> I can mark it as `#[doc(hidden)]` ok?

yep

> Also I think we could re-write the macro to get full failure case, such as Rust does it: https://github.com/rust-lang/rust/blob/master/src/libstd/sys/unix/mod.rs#L129-#L135

yeah, also it doesn't need to be a macro anymore, which would eliminate the need for `#[doc(hidden)]`